discovery-provider/src/utils/ipfs_lib.py

```diff
@@ -167,6 +167,43 @@ def cat(self, multihash):
             logger.error(f"IPFSCLIENT | IPFS cat timed out for CID {multihash}")
             raise  # error is of type ipfshttpclient.exceptions.TimeoutError
 
+    def multihash_is_directory(self, multihash):
+        try:
+            # attempt to cat single byte from CID to determine if dir or file
+            self._api.cat(multihash, 0, 1)
+            return False
+        except Exception as e:  # pylint: disable=W0703
+            if "this dag node is a directory" in str(e):
+                logger.warning(f'IPFSCLIENT | Found directory {multihash}')
+                return True
+
+        # Attempt to retrieve from cnode gateway endpoints
+        gateway_endpoints = self._cnode_endpoints
+        for address in gateway_endpoints:
+            gateway_query_address = "%s/ipfs/%s" % (address, multihash)
```

Comment:
urljoin?

Reply:
yes, this should become urljoin good catch

Reply:
nvm i dont think urljoin gracefully handles 3 or more parameters...not worth doing this now